website/project/views/register.py

```diff
@@ -104,6 +104,12 @@ def node_registration_retraction_post(auth, node, **kwargs):
             'message_long': 'Retractions of non-registrations is not permitted.'
         })
 
+    if node.root is not node:
```

Comment:
This check is made both here and in the model. Are both necessary, and if so, why?